zerver/tests/test_events.py

```diff
@@ -463,6 +464,29 @@ def test_change_realm_invite_required(self):
             error = schema_checker('events[0]', events[0])
             self.assert_on_error(error)
 
+    def test_change_realm_authentication_methods_enabled(self):
+        # type: () -> None
+        schema_checker = check_dict([
+            ('type', equals('realm')),
+            ('op', equals('update_dict')),
+            ('property', equals('default')),
+            ('data', check_dict([])),
+        ])
+        # Test every transition
+        for (auth_method) in \
+                ({'Google': True, 'Email': True, 'LDAP': True},
+                {'Google': True, 'Email': True, 'LDAP': False},
+                {'Google': True, 'Email': False, 'LDAP': False},
+                {'Google': True, 'Email': False, 'LDAP': True},
+                {'Google': False, 'Email': True, 'LDAP': True},
+                {'Google': False, 'Email': False, 'LDAP': True},
+                {'Google': False, 'Email': True, 'LDAP': False},
+                {'Google': False, 'Email': False, 'LDAP': False}):
```

Comment:
I think we probably don't want the full combinational explosion of `2**(number of auth backends)` -- I think we'd exercise all possible transitions if we did `TTT/FTT/TFT/TFF/TTT` (and wrote a comment explaining that any new backends should be just added with `T/T/T/F/T` as their sequence, so we'll always do exactly 5 transitions).